integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/massindexing/MassIndexingFailureIT.java

```diff
@@ -420,23 +456,48 @@ private void doMassIndexing(SessionFactory sessionFactory,
 			} );
 			backendMock.verifyExpectationsMet();
 		}
+		catch (AssertionError e) {
+			assertionError = e;
+			throw e;
+		}
 		finally {
 			Book.failOnBook2GetId.set( false );
 			Book.failOnBook2GetTitle.set( false );
+
+			if ( assertionError == null ) {
```

Comment:
I don't think `assertionError` may be different from null here. assertionError could have been changed only into the catch clause and the latter throws an exception. Maybe we can remove the check.

Reply:
The finally block gets executed whatever happens in catch blocks, even if they re-throw an exception. This `assertionError` may be null.